jobs/build-podman.Jenkinsfile

```diff
@@ -0,0 +1,379 @@
+                                // even when considering that all artifacts are around
+                                // the same size as those for x86_64
+                                env: ["OSBUILD_EXPORT_FORCE_NO_PRESERVE_OWNER":"true", "RUNVM_CACHE_SIZE": "25G"],
+                                expiration: "60m",
+                                image: params.COREOS_ASSEMBLER_IMAGE,
+                                workdir: WORKSPACE,
+                            )
+                            withEnv(["COREOS_ASSEMBLER_REMOTE_SESSION=${archinfo[arch]['session']}"]) {
+                                 shwrap("""cosa init --force https://github.com/coreos/fedora-coreos-config.git""")
+                            }
+                        }
+                    } else {
+                         shwrap("""cosa init --force https://github.com/coreos/fedora-coreos-config.git""")
+                    }
+                }]}
+            }
+
+            stage("Build Disk Images") {
+                  parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                        if (arch != "x86_64") {
+		            pipeutils.withOptionalExistingCosaRemoteSession(
+                                arch: arch, session: archinfo[arch]['session']) {
+                                generate_disk_images(artifacts, shortcommit, arch, params.CONTAINER_REGISTRY_STAGING_REPO)
+		            }
+                        } else {
+                            generate_disk_images(artifacts, shortcommit, arch, params.CONTAINER_REGISTRY_STAGING_REPO)
+                        }
+                   }]}
+	    }
+
+           stage("Kola") {
+               parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                   if (arch != "x86_64") {
+                       pipeutils.withOptionalExistingCosaRemoteSession(
+                           arch: arch, session: archinfo[arch]['session']) {
+                           shwrap("""cosa kola run  --basic-qemu-scenarios --qemu-image=./podman-${shortcommit}.qcow2""")
+                       }
+                   } else {
+                        shwrap("""cosa kola run  --basic-qemu-scenarios --qemu-image=./podman-${shortcommit}.qcow2""")
+                   }
+               }]}
+           }
+
+           stage("Compress Artifacts") {
+               parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                    if (arch != "x86_64") {
+                        def session =  archinfo[arch]['session']
+                        pipeutils.withOptionalExistingCosaRemoteSession(
+                        arch: arch, session: session) {
+                            parallel artifacts.collectEntries{artifact -> [artifact["platform"] , {
+                                def file = "podman-$arch-${shortcommit}.${artifact["suffix"]}"
+                                shwrap("""podman --remote exec -i $session /usr/bin/zstd -14 $file""")
```

Comment:
The canonical way to run arbitrary commands is `cosa shell --`. That'll work both in the remote and non-remote case (related to the previous comment about `withOptionalExistingCosaRemoteSession()`).